File: for-local-test/python/171.excel-sheet-column-number.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        result = 0
        for c in columnTitle:
            if c < 'A' or c > 'Z':
                return -1
            result = result * 26 + (ord(c) - ord('A') + 1)
        return result

if __name__ == "__main__":
    solution = Solution()
    columnTitle = "AB"
    print(f"Column number for '{columnTitle}': {solution.titleToNumber(columnTitle)}")  # Output: Column number for 'AB': 28

File: for-local-test/171.excel-sheet-column-number.java

class Solution {
    public int titleToNumber(String columnTitle) {
        int result = 0;
        for (char c : columnTitle.toCharArray()){
            if (c < 'A' || c > 'Z') return -1;
            result =  result * 26 + (c - 'A' + 1);
        }
        return result;
    }

    public static void main(String[] args){
        Solution sol = new Solution();
        String columnTitle = "AB";
        System.out.println(sol.titleToNumber(columnTitle));
    }
}


